webhooks/virtualmachine/validation/virtualmachine_validator.go

```diff
@@ -472,6 +476,42 @@ func (v validator) validateStorageClass(ctx *pkgctx.WebhookRequestContext, vm *v
 	return append(allErrs, field.Invalid(scPath, scName, fmt.Sprintf(storageClassNotAssignedFmt, vm.Namespace)))
 }
 
+func (v validator) validateCrypto(
+	ctx *pkgctx.WebhookRequestContext,
+	vm *vmopv1.VirtualMachine) field.ErrorList {
+
+	var (
+		allErrs      field.ErrorList
+		path         = field.NewPath("spec", "crypto", "className")
+		encClassName = vm.Spec.Crypto.EncryptionClassName
+	)
+
+	if encClassName == "" {
+		return nil
+	}
+
+	if !pkgcfg.FromContext(ctx).Features.BringYourOwnEncryptionKey {
+
+		allErrs = append(allErrs, field.Invalid(
+			path, encClassName, "feature not enabled"))
```

Comment:
This would allow users to specify `spec.crypto` fields (other than `encryptionClassName`) when the FSS is not enabled. Consider returning early if the `spec.crypto` is empty as well?

Reply:
Yes, because `crypto.useDefaultKeyProvider` is currently the default behavior for today. But yes, I should disallow setting that field to a non-nil value as well.